pkg/statistics/handle/autoanalyze/autoanalyze.go

```diff
@@ -392,6 +395,9 @@ func randomPickOneTableAndTryAutoAnalyze(
 
 		// We need to check every partition of every table to see if it needs to be analyzed.
 		for _, tbl := range tbls {
+			if !timeutil.WithinDayTimePeriod(start, end, time.Now()) {
```

Comment:
Could you please comment on why it is necessary to check it repeatedly?

Reply:
OK, I added.